src/compiler/core.ts

```diff
@@ -23,31 +23,151 @@ namespace ts {
         True = -1
     }
 
-    const createObject = Object.create;
-
     // More efficient to create a collator once and use its `compare` than to call `a.localeCompare(b)` many times.
     export const collator: { compare(a: string, b: string): number } = typeof Intl === "object" && typeof Intl.Collator === "function" ? new Intl.Collator() : undefined;
 
-    export function createMap<T>(template?: MapLike<T>): Map<T> {
-        const map: Map<T> = createObject(null); // tslint:disable-line:no-null-keyword
+    const createObject = Object.create;
+    /** Create a MapLike with good performance. Prefer this over a literal `{}`. */
+    export function createMapLike<T>(): MapLike<T> {

```

Comment:
If someone adds an enumerable property to `Object.prototype` that will cause problems for `{}`. Isn't that why we had this function in the first place?